docs/proposals/20211123-enhancement-of-cluster-networking.md

```diff
@@ -0,0 +1,306 @@
+In short, this solution use IPsec to build a VPN tunnel to provide layer 3 network connectivity among pods in different physical region,
+as there are in one vanilla kubernetes cluster.
+
+This proposal is inspired by [submariner](https://github.com/submariner-io/submariner), which is capable to connect multiple Kubernetes clusters.
+Comparing to multi-cluster architecture, OpenYurt follows single cluster architecture design, which reduce the complexity.
+
+## Motivation
+
+Edge-edge and edge-cloud communication is a common issue in edge computing scenarios.
+In OpenYurt, we have introduced YurtTunnel to cope with the challenges of O&M and monitoring in edge-cloud collaboration.
+
+YurtTunnel provides the capabilities for cloud nodes to execute kubectl exec/logs to edge nodes and to scrap metric from exporters on edge nodes.
+
+However, the problem that YurtTunnel solved is just a subset of edge-cloud communication, and currently there is no solution in OpenYurt for edge-edge communication.
+
+In some scenarios, pods in different physical region in OpenYurt cluster may need to talk to others using pod IP, service IP,
+or service domain name, as those pods are in a vanilla kubernetes cluster.
+
+This proposal aims to make it possible.
+
+### Goals
+
+- Design a network model of OpenYurt to achieve cross-nodepool communication.
+- Provide layer 3 network connectivity among pods across different nodepools, which means
+  pods in different nodepools can talk to each other by using pod IP, service IP and service domain name, as there 
+  are in one vanilla kubernetes cluster.
+- Replace the YurtTunnel.
+
+## Proposal
+
+The basic idea is as follows when connecting two nodepools:
+
+1. Set up a VPN tunnel between two nodes in each nodepool. Join all pod subnets(as known as podCIDR) into the VPN tunnel.
+   The nodes that hold the VPN tunnel is called gateway node.
+2. Configure route table, route policy, IP tables and other necessary network-related settings for nodes. Configure the gateway node as the next-hop of non-gateway nodes for routing packet to remote pod subnets.
+
+In the first implementation, I would like to choose [libreswan](https://github.com/libreswan/libreswan/) as the IPsec VPN implementation.
+
+When connecting two nodepools, the network topology is as follows:
+
+![networking-overview](../img/networking-overview.png)
+
+
+### New Components
+
+#### 1. YurtGateway
+A new component that is responsible for the VPN setup and teardown. The nodes YurtGateway deployed on is called gateway node.
+YurtGateway runs on specifically designated nodes in a nodepool. There will be only one active YurtGateway instance at a time in a nodepool.
+
+#### 2. YurtRouter
+A new component that is responsible for configuring route table, route policy and other network-related configurations on nodes.
+YurtRouter is a daemonset and is deployed on all nodepools that are participating in the VPN tunnel. 
+
+### Network Reachability Requirement
```

Comment:
This requirement is pretty challenge on edge side, Do you consider running a service on cloud to help establish the tunnel between different node-pools?

Reply:
@yixingjia edge yurt-gateway can connect with each other through cloud yurt-gateway. and the detail info will be added after next community meeting discussion.

Reply:
Ok, then the yurt-gateway on cloud can consider to change name like yurt-cloud-gateway and the gateway on the edge called yurt-edge-gateway. SDN have similar implementation for those kinds of requirements.

Reply:
@yixingjia we will discuss component name in the other proposal.